File: examples/lib/src/main/java/de/infolektuell/bass/LibLoader.java
package de.infolektuell.bass;

import com.un4seen.bass.Bass;

import java.io.InputStream;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardCopyOption;
import java.util.Objects;

final class LibLoader {
    private static boolean loaded = false;

    static synchronized void loadLibraries() {
        if (loaded) return;
        final String libName = System.mapLibraryName("bass");
        try (InputStream s = Bass.class.getResourceAsStream("/" + libName)) {
            if (Objects.isNull(s)) return;
            Path tmpDir = Files.createTempDirectory("bass");
            tmpDir.toFile().deleteOnExit();
            Path file = tmpDir.resolve(libName);
            Files.copy(s, file, StandardCopyOption.REPLACE_EXISTING);
            System.load(file.toString());
            loaded = true;
        } catch (Exception e) {
            loaded = false;
        }
    }
}

File: examples/lib/src/main/kotlin/de/infolektuell/bass/LibLoader.kt
package de.infolektuell.bass

import com.un4seen.bass.Bass
import java.nio.file.Files
import java.nio.file.StandardCopyOption

object LibLoader {
    private var loaded = false
    fun loadLibraries() {
        if (loaded) return
        val libName = System.mapLibraryName("bass")
        val osName = if (libName.endsWith(".dylib")) {
            "macos"
        } else if(libName.endsWith(".dll")) {
            "windows"
        } else  {
            "linux"
        }
        val archName = "x64"
        Bass::class.java.getResourceAsStream("/native/${osName}/${archName}/${libName}")?.let { s ->
            val tmpDir = Files.createTempDirectory("bass")
            val file = tmpDir.resolve(libName)
            Files.copy(s, file, StandardCopyOption.REPLACE_EXISTING)
            System.load(file.toString())
            loaded = true
            Files.delete(file)
        }
    }
}
